csr/csr.go

```diff
@@ -30,7 +30,8 @@ var badSignatureAlgorithms = map[x509.SignatureAlgorithm]bool{
 }
 
 // VerifyCSR checks the validity of a x509.CertificateRequest
```

Comment:
This now needs a comment indicating that it has a side effect of normalizing the input CSR. The comment should briefly explain what type of normalization happens.